File: programmers/LV2/테이블 해시 함수.cpp
#include <algorithm>
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int c = 0;

bool compare(vector<int> a, vector<int> b) {
    if(a[c] == b[c]) {
        return a[0] > b[0];
    } else {
        return a[c] < b[c];
    }
}

int solution(vector<vector<int>> data, int col, int row_begin, int row_end) {
    int answer = 0;
    c = col - 1;
    sort(data.begin(), data.end(), compare);
    
    vector<int> mod_temp;
    for(int i = row_begin; i <= row_end; i++) {
        int sum_temp = 0;
        for(int j = 0; j < data[0].size(); j++) {
            sum_temp += data[i - 1][j] % i;
        }
        mod_temp.push_back(sum_temp);
    }
    
    answer = mod_temp[0];
    for(int i = 1; i < mod_temp.size(); i++) {
        answer ^= mod_temp[i];
    }
    
    return answer;
}

File: programmers/LV2/테이블 해시 함수.swift
import Foundation

func solution(_ data: [[Int]], _ col: Int, _ row_begin: Int, _ row_end: Int) -> Int {
  let sortedData = data.sorted {
    if $0[col - 1] == $1[col - 1] {
      return $0[0] > $1[0]
    } else {
      return $0[col - 1] < $1[col - 1]
    }
  }

  var dataArr: [Int] = []
  for i in row_begin...row_end {
    let tuple = sortedData[i - 1]
    var modSum: Int = 0
    for t in tuple {
      modSum += t % i
    }
    dataArr.append(modSum)
  }

  return dataArr.reduce(0, ^)
}